connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1303,7 +1303,7 @@ private void reconfigureConnector(final String connName, final Callback<Void> cb
             }
             if (changed) {
                 List<Map<String, String>> rawTaskProps = reverseTransform(connName, configState, taskProps);
-                if (isLeader()) {
+                if (isLeader() || (leaderUrl() != null && leaderUrl().startsWith("NOTUSED"))) {
```

Comment:
This seems a little magic to me -- maybe add a comment on what is really happening here?

Reply:
Already commented.